compiler/stack_build.cmake

```diff
@@ -11,7 +11,7 @@
 cmake_policy (SET CMP0012 NEW)
 
 execute_process (
-    COMMAND ${STACK_EXECUTABLE} setup
+    COMMAND ${STACK_EXECUTABLE} setup --allow-different-user
```

Comment:
It looks to me that this is solving a problem specific to our use of Travis CI + a Docket container. It doesn't seem appropriate to pass `--allow-different-user` unconditionally here: for developer/customer machines that isn't the right default.

Is there a way we can fix this just from the Travis build script?

Reply:
That flag is no more needed.

Reply:
I've sent separate PRs #681 and #682.